storage/bucket.go

```diff
@@ -854,6 +888,62 @@ func (b *BucketAttrs) toRawBucket() *raw.Bucket {
 	}
 }
 
+func (b *BucketAttrs) toProtoBucket() *storagepb.Bucket {
+	// Copy label map.
+	var labels map[string]string
+	if len(b.Labels) > 0 {
+		labels = make(map[string]string, len(b.Labels))
+		for k, v := range b.Labels {
+			labels[k] = v
+		}
+	}
+
+	// Ignore VersioningEnabled if it is false. This is OK because
+	// we only call this method when creating a bucket, and by default
+	// new buckets have versioning off.
+	var v *storagepb.Bucket_Versioning
+	if b.VersioningEnabled {
+		v = &storagepb.Bucket_Versioning{Enabled: true}
+	}
+	var bb *storagepb.Bucket_Billing
+	if b.RequesterPays {
+		bb = &storage.Bucket_Billing{RequesterPays: true}
+	}
+	var bktIAM *storagepb.Bucket_IamConfig
+	if b.UniformBucketLevelAccess.Enabled || b.BucketPolicyOnly.Enabled || b.PublicAccessPrevention != PublicAccessPreventionUnknown {
+		bktIAM = &storagepb.Bucket_IamConfig{}
+		if b.UniformBucketLevelAccess.Enabled || b.BucketPolicyOnly.Enabled {
+			bktIAM.UniformBucketLevelAccess = &storagepb.Bucket_IamConfig_UniformBucketLevelAccess{
+				Enabled: true,
+			}
+		}
+		// TODO(noahdietz): This will be switched to a string.
+		//
+		// if b.PublicAccessPrevention != PublicAccessPreventionUnknown {
```

Comment:
When is this breaking change going to land? Wondering if we should hold off until then, since this is not that great to have to leave commented out.

Reply:
I don't have a concrete date, it is still in the works. I'd prefer not to block other work on this if possible, these adapters are necessary for implementing any gRPC methods.

Reply:
Sure, sounds good.